packages/x-date-pickers/src/models/adapters.ts

```diff
@@ -78,29 +78,36 @@ export type FieldFormatTokenMap = {
       };
 };
 
-export interface MuiPickersAdapter<TDate> {
+// https://www.zhenghao.io/posts/ts-never#how-to-check-for-never
+type PropertyIfNotNever<PName extends string, PType> = [PType] extends [never]
+  ? {}
+  : { [P in PName]?: PType };
+
+export type AdapterOptions<TLocale, TInstance> = {
+  formats?: Partial<AdapterFormats>;
+} & PropertyIfNotNever<'locale', TLocale> &
```

Comment:
Seems you never use `never` for locale
```suggestion
export type AdapterOptions<TLocale, TInstance> = {
  formats?: Partial<AdapterFormats>;
  locale?: TLocale;
} &
```

Reply:
It should be used on jalali and hijri moment adapters

Reply:
Fixed

Reply:
I can't get it to work with adapters without locales...
I'll accept `string` for moment jalali and hijri, that's not the end of the day I guess